src/test_stack.py

```diff
@@ -0,0 +1,95 @@
+"""This is the test for stack module."""
+import pytest
+import random
+
+PUSH_TABLE = [
+    [[1, 2, 3], 4, 4],
+    ["Foo to the bar", 5, 15],
+    [("a", "b", "c"), (5,), 4],
+    [{"one": 1, "two": 2}, {"three": 3}, 3],
+]
+
+POP_TABLE = [
+    [[1, 2, 3], 2],
+    ["Foo to the bar", 13],
+    [("a", "b", "c"), 2],
+    [{"one": 1, "two": 2}, 1],
+]
+
+PUSH_IT_TABLE = [
+    [4],
+    [5],
+    [(5,)],
+    [{"three": 3}],
+]
+
+PUSH_STACK_TABLE = [
+    [[1, 2, 3], 4],
+    ["Foo to the bar", 5],
+    [("a", "b", "c"), (5,)],
+    [{"one": 1, "two": 2}, {"three": 3}],
+]
+
+SIZE_DECREASES_TABLE = [
+    [7, 5],
+    [1000, 500],
+    [random.randint(75, 150), random.randint(0, 75)],
+    [1, 1]
+]
+
+@pytest.fixture
+def new_stack():
+    """Instantiate a new stack with no args."""
+    from stack import Stack
+    this_stack = Stack()
+    return this_stack
+
+
+@pytest.mark.parametrize("iterable, val, result", PUSH_TABLE)
+def test_push(iterable, val, result):
+    """Test the push method increases the size of the stack."""
+    from stack import Stack
```

Comment:
Since you guys went to the trouble of making a stack fixture, probably better to utilize it here right?